File: 23-01-2025/Missing Number.cpp
// https://leetcode.com/problems/missing-number/description/

//Code

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size(); 
        int total_sum = n * (n + 1) / 2;  
        int sum = 0;
        
        
        for(int num : nums) {
            sum += num;
        }
        
        return total_sum - sum;
    }
};


//Test Cases
/*
Input: nums = [3,0,1]
Output: 2

Input: nums = [0,1]
Output: 2

Input: nums = [9,6,4,2,3,5,7,0,1]
Output: 8
*/

File: 23-01-2025/java/MissingNumber.java
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int expectedSum = n*(n+1)/2;
        int actualSum=0;
        for(int i=0;i<nums.length;i++){
            actualSum+=nums[i];
        }
        return expectedSum - actualSum;
    }
    public  void main(String args[]){
        int nums[] = {9,6,4,2,3,5,7,0,1};
        System.out.println(missingNumber(nums));
    }
}